lib/class-wp-duotone-gutenberg.php

```diff
@@ -201,6 +201,43 @@ public static function output_footer_assets() {
 		}
 	}
 
+	/**
+	 * Adds the duotone SVGs and CSS custom properties to the editor settings so
+	 * they can be pulled in by the EditorStyles component in JS and rendered in
+	 * the post editor.
+	 *
+	 * @param array $settings The block editor settings from the `block_editor_settings_all` filter.
+	 * @return array The editor settings with duotone SVGs and CSS custom properties.
+	 */
+	public static function add_editor_settings( $settings ) {
+		$assets  = '';
+		$presets = 'body{';
+		foreach ( self::$global_styles_presets as $filter_data ) {
+			$assets  .= gutenberg_get_duotone_filter_svg( $filter_data );
+			$presets .= self::get_css_custom_property_declaration( $filter_data );
+		}
+		$presets .= '}';
+
+		if ( ! isset( $settings['styles'] ) ) {
+			$settings['styles'] = array();
+		}
+
+		$settings['styles'][] = array(
+			'assets'         => $assets,
+			'__unstableType' => 'svgs',
+			'isGlobalStyles' => false,
+		);
```

Comment:
Is `styles` the right place for `svg` markup? I'm wondering if it'd be worth having a different key in the block editor settings for it. I wasn't sure what the trade-offs might be between additional clarity by using a separate key, versus potentially having extra complexity if you then need to retrieve the `svg` markup from a different place.

I've added a similar, related comment on the JS code that filters out the svgs in `EditorStyles`.

Reply:
I don't know all the nuances with duotone, but I had similar thoughts with @andrewserong about having a different setting.